File: code/sdk/android/cloudsheet-Android-sdk/src/main/kotlin/com/opencloudsheet/response/onedrive/ListChildrenResponse.kt
package com.opencloudsheet.response.onedrive

import com.google.gson.annotations.SerializedName

data class ListChildrenResponse(
    @SerializedName("value")
    val value: List<DriveItem>,

    @SerializedName("@odata.nextLink")
    val odataNextLink: String? = null
)

data class DriveItem(
    @SerializedName("id")
    val id: String,

    @SerializedName("name")
    val name: String,

    @SerializedName("size")
    val size: Long? = null,

    @SerializedName("folder")
    val folder: FolderFacet? = null,

    @SerializedName("parentReference")
    val parentReference: ItemReference? = null
)

data class FolderFacet(
    @SerializedName("childCount")
    val childCount: Int? = null
)

data class ItemReference(
    @SerializedName("path")
    val path: String? = null
)

File: code/sdk/ios/OpenCloudSheet/Sources/OpenCloudSheet/Response/OneDrive/ListChildrenResponse.swift
//
//  ListChildrenResponse.swift
//  OpenCloudSheet
//

import Foundation

struct ListChildrenResponse: Codable {
    let value: [DriveItem]
    let odataNextLink: String?

    enum CodingKeys: String, CodingKey {
        case value
        case odataNextLink = "@odata.nextLink"
    }
}

struct DriveItem: Codable {
    let id: String
    let name: String
    let size: Int64?
    let folder: FolderFacet?
    let parentReference: ItemReference?
}

struct FolderFacet: Codable {
    let childCount: Int?
}

struct ItemReference: Codable {
    let path: String?
}
